nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/Instrumenter.java

```diff
@@ -92,40 +107,94 @@ private Instrumenter() {
     public static final void instrument(String rtJar, String outJar)
             throws NoSuchAlgorithmException, IOException {
         Logger.getGlobal().log(Level.FINE, "Instrumenting " + rtJar + " into " + outJar);
-        new Instrumenter().process(rtJar, outJar);
+        // rt.jar path will not be used in Java9+ environment
+        new Instrumenter().initAndProcess(rtJar, outJar);
     }
 
-    private void process(String rtJar, String outJar)
-            throws IOException, NoSuchAlgorithmException, FileNotFoundException {
-        ZipFile rt = null;
-        try {
-            rt = new ZipFile(rtJar);
-        } catch (IOException exc) {
-            Logger.getGlobal().log(Level.SEVERE, "Are you sure you provided a valid path to your rt.jar?");
-            throw exc;
+    private InputStream getClassInputStream(String className)
+            throws IOException {
+        InputStream clInputStream;
+        if (rtZipFile != null) {
+            try {
+                ZipEntry entry = rtZipFile.getEntry(className);
+                if (entry == null) {
+                    Logger.getGlobal().log(Level.WARNING, "Could not find " + className + " in rt.jar");
+                    Logger.getGlobal().log(Level.WARNING, "Are you running java 8?");
+                    Logger.getGlobal().log(Level.WARNING, "Continuing without instrumenting: " + className);
+                    return null;
+                }
+                clInputStream = rtZipFile.getInputStream(entry);
+                return clInputStream;
+            } catch (IOException exc) {
+                Logger.getGlobal().log(Level.WARNING, "Cannot find " + className + " are you sure this is a valid rt.jar?");
```

Comment:
There used to be a comment here about not knowing whether this code is reachable. Do you think it is for sure reachable? Otherwise, we may want to just keep the TODO around until we get to it.